### DISCUSSION TITLE
How to do migration & Seeding Database?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2220] [Q&A] [answered]
I am using pocketbase as framework and I need to know seeding database in two scenarios

1. During initializatin of pocketbase go migration initiated. After migration how to seed database?
2. Already existing instance new feature will be included through migration and it may require seeding only certain records to specific table for that update. How to achieve this?

**Comment by ganigeorgiev** [comment, marked as answer]
This is similar to your other question in https://github.com/pocketbase/pocketbase/discussions/2218.

The Go migrations are regular Go files, meaning that you can create a new Go migration with `migrate create` and then create the records you want to insert programmatically (aka. using the `models.NewRecord()`). For more info you can also check https://pocketbase.io/docs/migrations/#migrate-create.

If that doesn't help, please provide a little more detailed info what you are trying to do, how the seed data is loaded (is it from an external file or will be inside the migration), etc.

**Reply by iamdineshkumar** [reply]
Is that possible to create admin user during app initialization through migration? In migration documents only related record and settings examples are give. If admin user is created then its similar to manual installation done. How to achieve that?

**Reply by ganigeorgiev** [reply]
You can use any of the helpers from the [daos package](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.14.2/daos).

To create an admin inside a Go migration, you can do something like:

```go
dao := daos.New(db)

admin := new(models.Admin)
admin.Email = "test@example.com"
admin.SetPassword("123456789")

dao.SaveAdmin(admin)
```
